Chessington.GameEngine/Pieces/Pawn.cs

```diff
@@ -17,14 +17,29 @@ public override IEnumerable<Square> GetAvailableMoves(Board board)
             
             if (Player == Player.White)
             {
-                availableMoves.Add(new Square(currentPosition.Row - 1, currentPosition.Col));
-                if (currentPosition.Row == 6) availableMoves.Add(new Square(currentPosition.Row -2, currentPosition.Col));
+                if (!Square.At(currentPosition.Row - 1, currentPosition.Col).IsOccupied(board))
```

Comment:
I guess playing with white or black pieces is just a matter of a direction. Maybe you can identify a direction and do smth like `currentPosition.Row - 1*direction` ?

Reply:
This has been changed in the next pull request (10-12)